valve-monitor/README.md

```diff
@@ -0,0 +1,137 @@
+for supplying correct voltage to this project’s various components.
+* An I2C level shifter, such as [this one from HiLetgo](https://www.amazon.com/dp/B07F7W91LC/ref=cm_sw_r_api_i_FZPJ7VRY2329ARNQ3W22_0), to
+translate the flow meter’s pulses from 5V logic to 3.3V logic for the Swan
+microcontroller.
+* A MOSFET driver, such as this one from [HiLetgo](https://www.amazon.com/dp/B01I1J14MO/ref=cm_sw_r_api_i_8YRY25Q7R9HGV1ZPHERP_0),
+is required to allow the Swan to switch on high-power devices such as the
+solenoid, without needing to send that power from the microcontroller. (The
+Swan can only send about 0.066 Watts (3.3V*20mA) out of a GPIO.)
+
+Additionally you may wish to also use the following.
+
+* (Optional) A manual valve such as this [YZM Quick Connector](https://www.amazon.com/dp/B077H2JWSZ/ref=cm_sw_r_api_i_38J4S2VWFKAZBVEA8GCM_0)
+to make it easier to start/stop the flow of liquid while testing.
+* (Optional) An enclosure for your hardware such as this
+[outdoor-friendly enclosure from Sixfab](https://sixfab.com/product/raspberry-pi-ip54-outdoor-iot-project-enclosure/).
+
+Once you have all of your hardware you’ll next need to assemble the pieces.
+As a start, complete the [Notecard and Notecarrier-F quickstart guide](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/), which will help
+you connect your Notecard, Notecarrier, and Swan.
+
+![The Notecard, Notecarrier, and Swan connected](images/notecard-and-notecarrier.jpg)
+
+From there you’ll need to connect both your flow meter and solenoid valve
+to the Notecarrier. If you’re using this project’s recommended hardware, here’s a [wiring diagram that shows how all the components connect](https://miro.com/app/board/uXjVPL8v4hE=/).
+
+![Wiring diagram of the various valve monitor components](images/wiring-diagram.png)
+
+Regardless of which hardware you use, you’ll need to have the following pins
+wired to your Notecarrier.
+
+* The solenoid valve’s signal line must be connected to the Notecarrier’s `F_D6`
+pin. The Swan uses this pin to open and close the solenoid.
+* The flow meter’s signal line must be be connected to the Notecarrier’s `F_D10`
+pin. The Swan uses this pin to read the flow rate from the flow meter.
+* The Notecarrier’s `F_D13` pin must be wired to the Notecarrier’s `ATTN` pin. The
+Swan uses this to [respond to interrupts](https://dev.blues.io/guides-and-tutorials/notecard-guides/attention-pin-guide/)
+whenever the Notecard receives a command to open or close the solenoid valve.
+
+When everything is put together your build should look something like this.
+
+![The Notecarrier connected to the solenoid and the valve monitor](images/full-build-no-case.jpg)
+
+If you’re additionally using a case like the [recommended one from Sixfab](https://sixfab.com/product/raspberry-pi-ip54-outdoor-iot-project-enclosure/),
+your build can look like this when placed within the enclosure.
```

Comment:
Consider including a note about _gently_ bending the antenna and leaving it on the outermost edge of the case to ensure a good cellular connection.